lib/utils/appdb_reader.rb

```diff
@@ -55,22 +63,26 @@ def get_appliances(images)
         images.collect do |image|
           image['occi'] = [image['occi']].flatten.compact
           image['occi'].collect do |occi_image|
-            if occi_image.key? 'vo'
-              voname = occi_image['vo']['name']
-              image_id = occi_image['id'].split('os_tpl#')[1]
-            else
-              image_id = ''
-              voname = ''
-            end
-            {
-              id: image_id,
-              name: image['application']['name'],
-              mpuri: image['mpuri'],
-              vo: voname
-            }
+            format_image(image, occi_image)
           end
         end.flatten
       end
+
+      def format_image(image, occi_image)
+        if occi_image.key? 'vo'
+          voname = occi_image['vo']['name']
```

Comment:
It definitely might, but "standard" for appdb response describes, that it cannot happen - if you know which standard i mean ;-). Same comment can be used for multiple fields.